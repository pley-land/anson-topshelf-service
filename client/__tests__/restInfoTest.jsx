import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RestInfo from '../components/restInfo';

describe('Testing RestInfo', () => {
  const wrapper = shallow(<RestInfo name="in-n-out" averageRating={4} reviews={995} price={2} tags="Chinese, Japanese" />);
  it('should render a rating component', () => {
    expect(wrapper.find('#ratings').length).toEqual(1);
  });
  it('should render the restaurant name', () => {
    expect(wrapper.find('#rest-name-title').length).toEqual(1);
  });
  it('should render the number of reviews', () => {
    expect(wrapper.find('#reviews').length).toEqual(1);
  });
  it('should render a details component', () => {
    expect(wrapper.find('#details').length).toEqual(1);
  });
  it('should render a price range', () => {
    expect(wrapper.find('#price-range').length).toEqual(1);
  });
  it('should render a tags component', () => {
    expect(wrapper.find('#rest-tags').length).toEqual(1);
  });
  const output = renderer.create(<RestInfo tags="Chinese, Japanese" />).toJSON();
  it('should render the whole component', () => {
    expect(output).toMatchSnapshot();
  });
});
