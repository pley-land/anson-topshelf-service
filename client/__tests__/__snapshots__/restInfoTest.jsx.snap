// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RestInfo should render the whole component 1`] = `
<div
  className="content-rest-info"
>
  <h1
    id="rest-name-title"
  />
  <div
    className="rest-info-numbers"
  >
    <div
      id="ratings"
    >
      <div
        className="starDiv-empty"
      >
        <i
          className="material-icons stars-empty icon-white stars"
        >
          star
        </i>
      </div>
      <div
        className="starDiv-empty"
      >
        <i
          className="material-icons stars-empty icon-white stars"
        >
          star
        </i>
      </div>
      <div
        className="starDiv-empty"
      >
        <i
          className="material-icons stars-empty icon-white stars"
        >
          star
        </i>
      </div>
      <div
        className="starDiv-empty"
      >
        <i
          className="material-icons stars-empty icon-white stars"
        >
          star
        </i>
      </div>
      <div
        className="starDiv-empty"
      >
        <i
          className="material-icons stars-empty icon-white stars"
        >
          star
        </i>
      </div>
    </div>
    <div
      id="reviews"
    >
      undefined reviews
    </div>
    <div
      id="details"
    >
      <a
        href=""
      >
        Details
      </a>
    </div>
  </div>
  <div
    id="rest2"
  >
    <div
      id="price-range"
    />
    <div
      id="rest-tags"
    >
      <a
        href="https://www.google.com/search?q=Chinese+cuisine"
      >
        Chinese, 
      </a>
      <a
        href="https://www.google.com/search?q=Japanese+cuisine"
      >
        Japanese
      </a>
    </div>
  </div>
</div>
`;
